From c3d6ad34ee84a32e7ba8a4832f286643f452d62e Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Fri, 19 Jan 2024 13:12:25 +0100
Subject: [PATCH 0640/1695] plat-stm32mp2: conf: add hse monitoring support on
 stm32mp25

Add the possibility to detect HSE over frequency detection
on STM32MP25 devices.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git db1789e84a55237145ef8c6d08de87cee5b31b92 plat-stm32mp2: conf: add hse monitoring support on stm32mp25]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: Ic70aa0da61cad057c34887061218adf2bc138616
---
 core/arch/arm/plat-stm32mp2/conf.mk | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index b4cc6a613..2d8a8468d 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -82,6 +82,7 @@ $(call force,CFG_SCMI_SERVER_REGULATOR_CONSUMER,y)
 $(call force,CFG_SECURE_TIME_SOURCE_CNTPCT,y)
 $(call force,CFG_STM32_SHARED_IO,y)
 $(call force,CFG_STM32_FIREWALL,y)
+$(call force,CFG_STM32_HSE_MONITORING,y)
 $(call force,CFG_STM32_PWR,y)
 $(call force,CFG_STM32_PWR_REGUL,y)
 $(call force,CFG_STM32MP_CLK_CORE,y)
@@ -167,6 +168,10 @@ CFG_STM32_EARLY_CONSOLE_UART ?= 2
 # Default disable external DT support
 CFG_EXTERNAL_DT ?= n
 
+ifeq ($(CFG_STM32_HSE_MONITORING),y)
+$(call force,CFG_STM32_LPTIMER,y)
+endif
+
 CFG_STM32MP_REMOTEPROC ?= n
 CFG_DRIVERS_REMOTEPROC ?= $(CFG_STM32MP_REMOTEPROC)
 CFG_REMOTEPROC_PTA ?= $(CFG_STM32MP_REMOTEPROC)
-- 
2.47.2

